.github/workflows/python_version_compatibility.yaml

```diff
@@ -0,0 +1,21 @@
+name: Check python version compatibility
+
+on:
+  push:
+    branches:
+      - develop
+  workflow_dispatch:
+
+
+jobs:
+  pythonversionsmatrix:
+    strategy:
+      matrix:
+        python_version: [3.8, 3.9, 3.10, 3.11, 3.12, 3.x]
```

Comment:
don't you only select 3.8 upwards? Where is 3.6 selected? And what does 3.x do here? Is this selecting the latest version? 

I'm just trying to understand the syntax.

Reply:
See [here](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) for what 'matrix' does - essentially it takes all possible combinations of python_version and os, plus those given by `include` (python 3.6 is not available via github actions on ubuntu 22.04, so I've made it run on 20.04 instead).